docs/runtimes/functions.md

```diff
@@ -1,27 +1,73 @@
+Once you register a function within a project, you can execute it in your local 
+environment or at scale on a Kubernetes cluster.
+
+The relationship between **functions**, [**workflows**](../projects/build-run-workflows-pipelines.html), and [**projects**](../projects/project.html), is as follows:
+
+![MLRun Function](../_static/images/mlrun_concepts_architecture.png)
+
+After the MLRun functions and workflows are created and [**registered into the project**](../projects/create-project.html), they are invoked using the project object. 
+This workflow pairs especially well with {ref}`ci-cd-automate`.
+		
+## More about functions	
+
+There are various kinds of MLRun functions with different capabilities, however, 
+there are commonalities across all functions. In general, an MLRun function looks like the following:
+
+<img src="../_static/images/mlrun-functions.png" alt="mlrun-architecture" width="700"/><br>
 
 MLRun supports:
-- Real-time functions for: {ref}`serving <serving-graph>`, APIs, and stream processing (based on the high-performance {ref}`Nuclio <nuclio-real-time-functions>` engine). 
-- Batch functions (based on Kubernetes jobs, {ref}`Spark <spark-operator>`, {ref}`Dask <dask-overview>`, {ref}`Horovod <horovod>`, etc.)
+- Real-time runtimes for: {ref}`serving <serving-graph>`, APIs, and stream processing (based on the high-performance {ref}`Nuclio <nuclio-real-time-functions>` engine). 
+- Batch runtimes (based on Kubernetes jobs, {ref}`Spark Operator<spark-operator>`, {ref}`Dask <dask-overview>`, {ref}`Horovod <horovod>`, etc.)
+
+The [different function runtimes](../concepts/functions-overview.html) take care of automatically transforming the code and spec to fully 
+managed and elastic services over Kubernetes, which saves significant operational overhead, 
+addresses scalability, and reduces infrastructure costs. The function parameters and capabilities are explained in more detail in {ref}`create-and-use-functions`.
 
-Function objects are all inclusive (code, spec, API, and metadata definitions), which allows placing them 
-in a shared and versioned function market place. This means that different members of the team can produce or 
-consume functions. Each function is versioned and stored in the MLRun database with a unique hash code, 
+Function objects are all inclusive, containing the code and all the operational aspects: ({ref}`image <images-usage>`, required packages, 
+{ref}`pod resource congifuration (replicas, CPU/GPU/memory limits, volumes, Spot vs. On-demand nodes, pod priority, node affinity) <configuring-job-resources>`, 
+[storage](../runtimes/function-storage.html), environment, metadata definitions, etc.). Each function is versioned and stored in the MLRun database with a unique hash code, 
 and gets a new hash code upon changes.
 
+When you create a function, you specify its name, which appears in the MLRun UI, as well as the Kubernetes pod.
```

Comment:
This sentence is not clear - it may be understood from it that you specify its k8s pod, which not only is not correct, it also negates what's written above.
I assume you meant that the the k8s pod used to execute the code appears in the MLRun UI, but why is this relevant here? This sentence deals with what happens when you create a function.

Reply:
deleted